### PR TITLE
Pipeline changes - github actions and Helm charts

### PR DESCRIPTION
**Pull request by TimCsaky**
# Description

Replace CI/CD with github actions, Helm charts

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://dgrsc-dev.apps.silver.devops.gov.bc.ca/pr-87

File: charts/dgrsc/templates/deploymentconfig.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            - name: SERVER_KC_CLIENTSECRET
+              valueFrom:
+                secretKeyRef:
+                  key: password
+                  name: {{ include "dgrsc.configname" . }}-keycloak-secret
+            - name: CDOGS_CLIENTSECRET
+              valueFrom:
+                secretKeyRef:
+                  key: password
+                  name: cdogs-service-secret
```
Comment (jujaga) [inline charts/dgrsc/templates/deploymentconfig.yaml:72]: For production-grade secrets, we should not be disclosing either the username/clientId as a part of the embedded config map. The old deployment config had the client id being pulled from the secret - this pattern should remain upheld.

File: charts/dgrsc/values.yaml [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    SERVER_APIPATH: "/api/v2"
+    SERVER_BASEPATH: "/app"
+    SERVER_BODYLIMIT: "30mb"
+    # SERVER_HOSTURL: ~
+    # SERVER_LOGFILE: ~
+    SERVER_LOGLEVEL: "http"
+    SERVER_PORT: "8080"
+    # CDOGS_APIURL: ~
+    # CDOGS_CLIENTID: ~
+    # CDOGS_TOKENURL: ~
```
Comment (jujaga) [inline charts/dgrsc/values.yaml:133]: These are commented meaning they are optional in configMap setup. We shouldn't be having the username/clientId side of the secret being publicly visible on a github repository.

**Review by jujaga — APPROVED** [review]

File: charts/dgrsc/templates/deploymentconfig.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            - name: SERVER_KC_CLIENTID
+              valueFrom:
+                secretKeyRef:
+                  key: username
+                  name: app-keycloak-secret
+            - name: SERVER_KC_CLIENTSECRET
+              valueFrom:
+                secretKeyRef:
+                  key: password
+                  name: app-keycloak-secret
```
[the remark this answers is not in the document]
Comment (TimCsaky) [reply charts/dgrsc/templates/deploymentconfig.yaml]: changed to `document-generation-showcase-keycloak-secret`

**Pull request merged**